network/src/gossip/mod.rs

```diff
@@ -0,0 +1,1206 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
```

Comment:
Nit: this file could have stayed `network/src/gossip.rs` (with Rust edition 2018 it's the idiomatic way), to make for a readable diff.

Reply:
What about it makes it more readable?
It's strange to me that this is considered idiomatic. Before, you either had all logic for a module in one file or one directory. Now it can be spread all over the place.

Reply:
If refactoring gossip out to a separate module was done in another PR it would make these changes much more readable. rn it's very hard to tell what has changed and what stays the same since everything has changed.